launcher/src/main/java/org/apache/spark/launcher/ChildProcAppHandle.java

```diff
@@ -112,7 +112,7 @@ void monitorChild() {
         }
       }
 
-      disconnect();
+      dispose();
```

Comment:
Can we add more document to `disconnect` and `dispose`? So that people can understand the difference between them clearly and have a better understanding of changes like this.

Reply:
`disconnect()` is actually a public method and already documented in the SparkAppHandle interface.

Reply:
Sorry I'm still not able to figure out what's the difference between them after reading the doc, do you mind leave a short description here? thanks!

Reply:
I updated the documentation for dispose.